FreshRSS/FreshRSS
# Standardise command line option parsing

# Changes proposed in this pull request:

Continuing on from the work done in #6028, parsing of options entered on the command line is now standardised across all relevant commands (those that accept options) giving us a single source of truth and potentially clearing the way to further standardisation.

Additionally: we now validate short as well as long options, all existing short options now have a long equivalent, and it would be trivial to add short equivalents for any existing long options. Relevant documentation (e.g. the cli README as well as any --help or usage outputs) has been updated to take account of these changes.

I've also scooped up a fix to check.translation.php which was failing to run due to not having access to the I18N_PATH constant and now runs as expected.

# How to test the feature manually:

1. To see short option validation use an unknown short option with any command e.g. 
```shell
cli/do-install.php --default-user admin -q
```
Previous behaviour: 
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/5b13bec7-1f72-44bd-bb24-2826641f7e1a)

New behaviour:
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/8289c919-a663-4729-8092-8f88f43e4938)

2. All other behaviour remains unchanged and so can be tested by using commands and verifying that the result is as would have previously been expected e.g.
```shell
cli/user-info.php --user foo --json
```
will still provide information for user 'foo' in json format etc.

# Pull request checklist:
